integration_tests/steps/two_sided_pay_as_clear.py

```diff
@@ -32,6 +32,18 @@ def test_traded_energy_rate(context):
                                         for rate in d.clearing_rate])
 
 
+@then('buyers and sellers are not same')
+def test_different_buyer_seller(context):
+    grid = context.simulation.area
+    for child in grid.children:
```

Comment:
Please remove the nested for loops as it makes it unreadable, you do not need to access the markets via the IAA class, you can achieve the same result via the area classes.